build-cran-package.sh

```diff
@@ -151,9 +151,8 @@ cd "${TEMP_R_DIR}"
         -e 's/^.*#pragma region.*$//' \
         -e 's/^.*#pragma endregion.*$//' \
         -e 's/^.*#pragma warning.*$//' \
-        "${file}"
-    done
-    find . -name '*.h.bak' -o -name '*.hpp.bak' -o -name '*.cpp.bak' -exec rm {} \;
+        {} +
+    find . \( -name '*.h.bak' -o -name '*.hpp.bak' -o -name '*.cpp.bak' \) -exec rm {} \;
```

Comment:
```suggestion
    find . -name '*.bak' -exec rm {} \;
```

While you're touching this anyway, let's just change this to deleting all `.bak` files from this directory and below. None of them are necessary, they're all just temporary side effects of the `sed` usage.

Reply:
and actually.... could you move this down to where it says "Cleaning sed backup files", replacing this `rm`?

https://github.com/microsoft/LightGBM/blob/41ba9e8f00c89d72e5cb71c964722ce1ed4d8445/build-cran-package.sh#L167

Might as well just do that only 1 time and catch all of them.